lib/github/ldap/member_search/recursive.rb

```diff
@@ -34,73 +34,72 @@ def initialize(ldap, options = {})
         #
         # Returns Array of Net::LDAP::Entry objects.
         def perform(group)
-          found = Hash.new
-
-          # find members (N queries)
-          entries = member_entries(group)
-          return [] if entries.empty?
-
-          # track found entries
-          entries.each do |entry|
-            found[entry.dn] = entry
+          found    = Hash.new
+          searched = []
+          entries  = []
+
+          # if this is a posixGroup, return members immediately (no nesting)
+          uids = member_uids(group)
+          return entries_by_uid(uids) if uids.any?
+
+          # track group
+          searched << group.dn
```

Comment:
From what I can tell, `searched` tracks DNs for groups, members, and subgroups. Maybe a better explanation or variable name here?